scripts/customEventHooks.lua

```diff
@@ -2,64 +2,194 @@ local customEventHooks = {}
 
 customEventHooks.validators = {}
 customEventHooks.handlers = {}
+customEventHooks.scriptID = {
+	handlers = {},
+	validators = {},
+	generatedScriptIDs = {}
+}
+
+function customEventHooks.generateScriptID(filePath)
+	if not string.match(filePath, "%S") then 
+		return nil
+	end
+
+	if customEventHooks.scriptID.generatedScriptIDs[filePath] then
```

Comment:
This is just caching because else we generate a fresh ID every time we add a handler, its designed to have the same ID but multiple validators and handlers registered to one scriptID